In [ ]:
print("Merhaba, İGÜ - İSTKA / Ders 7") # print ekrana çıktı veren bir koddur.


Merhaba, İGÜ - İSTKA / Ders 7


In [ ]:
import sys # import python programlama dilinde yazılıma kütüphane eklemeye yarayan bir kod blogudur.
print(sys.version)

3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]


In [2]:
from google.colab import drive # google colab - drive bağlantısı

drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import os

os.chdir('/content/drive/My Drive/VERIYAPILARI_ISTKA')

!pwd # bize kodun çıktısını dosya yolu olarak veren kod blogudur.

/content/drive/My Drive/VERIYAPILARI_ISTKA


In [ ]:
# Kullanıcıdan kısa kenar değerini al
while True:
    try:
        kenar1 = int(input("Kısa Kenar Degerini Girin "))
        if kenar1 < 0:
            kenar1 = abs(kenar1)
            print(f"Negatif değer girdiniz, pozitif olarak alındı: {kenar1}")
        break
    except ValueError:
        print("Lütfen geçerli bir tam sayı girin.")

# Kullanıcıdan uzun kenar değerini al
while True:
    try:
        kenar2 = int(input("Uzun Kenar Degerini Girin: "))
        if kenar2 < 0:
            kenar2 = abs(kenar2)
            print(f"Negatif değer girdiniz, pozitif olarak alındı: {kenar2}")
        break
    except ValueError:
        print("Lütfen geçerli bir tam sayı girin.")


def dikdortgen_alani(kenar1, kenar2):
    return kenar1 * kenar2

alan = dikdortgen_alani(kenar1, kenar2)
print("Kenarları ", kenar1 , " ve ", kenar2 , " olan dikdörtgenin alanı: ", alan)


Kısa Kenar Degerini Girin 5
Uzun Kenar Degerini Girin: 6
Kenarları  5  ve  6  olan dikdörtgenin alanı:  30


QUIZ 2- 1 ile 100 arasındaki sayılardan 12'ye bölünenleri bulan Python kodu

In [ ]:
Sayi_dizisi = []
sayi = 1

while sayi <= 175:
    if sayi % 12 == 0:
        Sayi_dizisi.append(sayi)
    sayi += 1

# Sonucu ekrana yazdır
print("1 ile 100 arasındaki 12'ye bölünen sayılar:", Sayi_dizisi)
print("Toplam sayı adeti: ", len(Sayi_dizisi))


1 ile 100 arasındaki 12'ye bölünen sayılar: [12, 24, 36, 48, 60, 72, 84, 96, 108, 120, 132, 144, 156, 168]
Toplam sayı adeti:  14


1 ile 100 arasındaki 12'ye bölünen sayılar: [12, 24, 36, 48, 60, 72, 84, 96, 108, 120, 132, 144, 156, 168]
Toplam sayı adeti:  14
PYTHON PROGRAMLAMA DİLİNDE DOSYA İŞLEMLERİ

Python programlama dilinde dosya işlemleri; dosyaların okunması, yazılması, güncellenmesi ve kapatılması gibi işlemleri kapsamaktadır. Dosya işlemleri, genellikle dosyaların içeriğini almak, bilgi üzerinde işlem yapabilmek veya yeni dosyalar oluşturmak için kullanılır.

PYTHON programlama dili üzerindeki dosya işlemleri bağlamında Yapay Zeka (YZ) çalışmaları noktasında veri setlerini ele aldık. Veri yapıları ve Algoritmalar'ın sonucu olarak yapay zeka çalışmaları ele alacağız. Doğru veri yapısı için bu bağlamda yapay zeka uygulamaları için veri setlerinin önemi:

Modelin Doğruluğu

Geniş Kapsamlı bir genelleme

Performans

Modilin başarısı için verilerin doğruluğu çok
önemlidir.

In [ ]:
import pandas as pd

VeriSetiYolu = '/content/drive/My Drive/VERIYAPILARI_ISTKA/veriseti'

for file in os.listdir(VeriSetiYolu):                         # Klasördeki tüm dosyaları listele

      if file.endswith(('.xlsx', '.xls')):                     # Sadece Excel dosyalarını seç

          df = pd.read_excel(os.path.join(VeriSetiYolu, file))  # Excel dosyasını pandas ile oku

          df.to_csv(os.path.join(VeriSetiYolu, file.rsplit('.', 1)[0] + '.csv'), index=False)
                                                                 # DataFrame'i CSV formatında kaydet
          print(f"{file} dosyası CSV formatına çevrildi.")     # Çevrim İşlemi Çıktısı

  # İşlem tamamlandığında kaç tane excel dosyası var ise o kadar *.csv formatına çevildiğinin
  # bilgisini almaktayız

atktan-geri-kazanm-miktarlar.xlsx dosyası CSV formatına çevrildi.


In [3]:
import os
import pandas as pd

VeriSetiYolu = '/content/drive/My Drive/VERIYAPILARI_ISTKA/veriseti'

# VeriSetiYolu içindeki .csv dosyalarını bul ve say
csv_dosyalari = [dosya for dosya in os.listdir(VeriSetiYolu) if dosya.endswith('.csv')]

# .csv dosya sayısını ekrana yazdır
csv_dosya_sayisi = len(csv_dosyalari)
print(f"VeriSetiYolu içerisinde {csv_dosya_sayisi} tane .csv dosyası bulunuyor.")

# Mevcut *.csv uzantılı dosyaların isimlerini getiriyoruz.
print("Bulunan CSV Dosyaları:")
for dosya in csv_dosyalari:
    print(dosya)


VeriSetiYolu içerisinde 2 tane .csv dosyası bulunuyor.
Bulunan CSV Dosyaları:
atktan-geri-kazanm-miktarlar.csv
atık.csv
